extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,197 @@
+  if (is.na(user_guid)) {
+    # NA user_guid populates when ACL is no login required / public access
+    as.character("Anonymous")
+  } else {
+    user_endpoint <- paste0("v1/users/", user_guid)
+    user <- client$GET(user_endpoint)
+    user$username
+  }
+}
+
+# checks to see if a content item has failed jobs, grabs usage data if it does,
+# then compiles content, job, and usage data together, returning it.
+get_failed_job_data <- function(item, usage, client, connectServer) {
+  failed_jobs <- tryCatch(
+    {
+      get_jobs(item) |> 
+        # filter successful jobs
+        filter(exit_code != 0) |> 
+        # map content job types to something more readable 
+        mutate(tag = case_when(
+               tag == "unknown" ~ "Unknown Job Type",
+               tag == "build_report" ~ "Build Report",
+               tag == "build_site" ~ "Build Site",
+               tag == "build_jupyter" ~ "Build Jupyter Notebook",
+               tag == "packrat_restore" ~ "Packrat Cache Restore",
+               tag == "python_restore" ~ "Python Envrionment Restore",
+               tag == "configure_report" ~ "Parametrized Report Configure",
+               tag == "run_app" ~ "Run R Application",
+               tag == "run_api" ~ "Run Plumber API",
+               tag == "run_tensorflow" ~ "Run Tensorflow API Model",
+               tag == "run_python_api" ~ "Run Python API",
+               tag == "run_dash_app" ~ "Run Dash Application",
+               tag == "run_gradio_app" ~ "Run Gradio Application",
+               tag == "run_streamlit" ~ "Run Streamlit Application",
+               tag == "run_bokeh_app" ~ "Run Bokeh Application",
+               tag == "run_fastapi_app" ~ "Run Python FastAPI",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "render_shiny" ~ "Render Quarto or RMarkdown Interactive Document",
+               tag == "run_voila_app" ~ "Run Voila Application",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "ctrl_extraction" ~ "Extract Parameter Controls From Report",
+               TRUE ~ tag)) |>
+        # map exit codes to something more readable 
+        mutate(exit_code = as.character(exit_code)) |>
+        mutate(exit_code = case_when(
+               exit_code == "1" ~ "Ended with errors",
+               exit_code == "127" ~ "Command could not be executed outside container",
+               exit_code == "137" ~ "Application terminated due to low memory",
+               exit_code == "2" ~ "Command encountered an error during execution",
+               exit_code == "134" ~ "Terminated due to critical error, potential segfault",
+               exit_code == "255" ~ "Abnormal termination of process",
+               exit_code == "130" ~ "Process terminated by user-initiated signal",
+               exit_code == "13" ~ "File permissions issue",
+               exit_code == "15" ~ "Process terminated externally",
```

Comment:
Could we collapse some of these into ones that are more user-focused? I would think that something like: 1, 2, and 134 are all a group of "failed to run / error during running"; 137 is "out of memory"; 255, 130, and 15 are "process terminated by server"; and 13 and 127 are suggestive of configuration errors? Those are slightly more actionable for users (or at least that first and second group are, the others are "talk to your sysadmin")